incuna_auth/middleware/login_required.py

```diff
@@ -38,40 +36,35 @@ class LoginRequiredMiddleware:
     Will default to protecting everything if LOGIN_PROTECTED_URLS is not in
     settings.
 
-
     Original code from:
     http://onecreativeblog.com/post/59051248/django-login-required-middleware
 
     This version has been modified to allow us to define areas of the site to
     password protect instead of protecting everything under /.
     '''
-    def process_request(self, request):
-        assert hasattr(request, 'user'), (
-            "The 'LoginRequiredMiddleware' "
-            + "requires authentication middleware to be installed. Edit your "
-            + "'MIDDLEWARE_CLASSES' setting to insert "
-            + "'django.contrib.auth.middlware.AuthenticationMiddleware'. "
-            + "If that doesn't work, ensure your 'TEMPLATE_CONTEXT_PROCESSORS' "
-            + "setting includes 'django.core.context_processors.auth'."
-        )
-
-        # Jump over this middleware if not a protected url.
-        path = request.path_info.lstrip('/')
-        path_is_exempt = any(m.match(path) for m in EXEMPT_URLS)
-        path_is_protected = any(m.match(path) for m in PROTECTED_URLS)
-        if path_is_exempt or not path_is_protected:
-            return
-
-        # Jump over this middleware if user logged in.
-        if request.user.is_authenticated():
-            return
+    """
+    login_exempt_urls = getattr(settings, 'LOGIN_EXEMPT_URLS', [])
+    login_protected_urls = getattr(settings, 'LOGIN_PROTECTED_URLS', [r'^'])
 
-        # Raise a 403 for POST/DELETE etc.
-        if request.method != 'GET':
-            return HttpResponseForbidden()
+    EXEMPT_URLS = [compile_url(settings.LOGIN_URL), compile_url(settings.LOGOUT_URL)]
+    EXEMPT_URLS += compile_urls(login_exempt_urls)
+    PROTECTED_URLS = compile_urls(login_protected_urls)
 
-        # Add a message, and redirect to login.
-        if SEND_MESSAGE:
-            messages.info(request, _('You must be logged in to view this page.'))
+    def process_request(self, request):
+        self.assert_request_has_user(request)
+
+        # Jump over this middleware if the page doesn't come under its jurisdiction.
+        if not self.is_url_protected(
+            request,
+            feincms_states=[AccessState.STATE_AUTH_ONLY],
+            exempt_urls=self.EXEMPT_URLS,
+            protected_urls=self.PROTECTED_URLS
+        ):
+            return
 
-        return redirect_to_login(request.path_info)
+        # Deny access if the user is not logged in.
+        if request.user.is_anonymous():
+            error = ''
+            if SEND_MESSAGE:
+                error = 'You must be logged in to view this page.'
+            return self.deny_access(request, error)
```

Comment:
That's my point; you're just going to override `process_request` every single time, so there's not much point adding it (or testing it), I think.

The check sounds kind of neat, although `The system check framework is a set of static checks` suggests it's not useful here since the `request`s are built dynamically.